docs/packages/context-menu/Context-menu-manager.md

```diff
@@ -0,0 +1,345 @@
+# Context Menu Manager (API)
+
+### Table of content
+- [Introduction](#Introduction)
+    - [Context menu UI behavior](#Context menu UI behavior)
+    - [Context-Menu config structure](#Context Menu config structure)
+    - [Structure tree](#Structure tree)
+- [API methods]()
+  - [Create a group](#Create a group)
+  - [Update a group](#Update a group)
+  - [Delete a group](#Delete a group)
+  - [Add item](#Add item)
+  - [Update item](#Update item)
+  - [Delete item](#Delete item)
+  - [Get config](#Get config)
+- [Understanding the usage flow]()
+  - [Registration of new items](#Registration of new items)
+  - [Triggering the context menu flow](#Triggering the context menu flow)
+- [Example of usage]()
+
+## Introduction
+The context menu manager provides a convenient way for managing the context menu structure.
+The manager is responsible for building a tree of nested groups/items that forms the actual structure of the context menu, as presented in the UI.   
+The manager is based on CRUD principles while providing simple functions for adding, updating, deleting, and in the end - provides a function for reading the main tree that have been created.
+
+### Context menu UI behavior
+For better understanding the structure and the manager, lets see some important UI notes:
+
+* The context menu displays a list of items, by default each item provides an action when clicking on it.
+
+* Items can provide action also on hover (mouseEnter).
+
+* If an item contains children groups, hovering on it will open a sub menu, and clicking on it won't trigger action.
+
+* Groups themselves are not visually displayed, they just displays their items and separate between them.
+
+* A group can display a title, if a title is specified.
+
+* Items of a group are displayed together, separated from other group items with a line separator.
+
+* Nested groups are displayed as a sub-menu aside the main context-menu.
+
+### Context Menu config structure
+The context menu config structure is built as a tree of groups & items.
+The root is based on groups, each group contains one or more item (inside its `items[]` property).
+Items can also have children, inside their `groups[]` property. those children are *nested groups*.
```

Comment:
```suggestion
Items can also have children, inside their `groups` property.
```

I don't think it's necessary

Reply:
Partly axepted.

Reply:
It's crucial to understand the structure.

Reply:
yeah, but it's obvious that a group that is a child of an item is a *nested* group...